src/pages/home/report/withReportOrNotFound.tsx

```diff
@@ -0,0 +1,81 @@
+import React, {ComponentType, ForwardedRef, RefAttributes} from 'react';
+import {OnyxEntry, withOnyx} from 'react-native-onyx';
+import {RouteProp} from '@react-navigation/native';
+import getComponentDisplayName from '../../../libs/getComponentDisplayName';
+import NotFoundPage from '../../ErrorPage/NotFoundPage';
+import ONYXKEYS from '../../../ONYXKEYS';
+import FullscreenLoadingIndicator from '../../../components/FullscreenLoadingIndicator';
+import * as ReportUtils from '../../../libs/ReportUtils';
+import * as OnyxTypes from '../../../types/onyx';
+
+type OnyxProps = {
+    /** The report currently being looked at */
+    report: OnyxEntry<OnyxTypes.Report>;
+    /** The policies which the user has access to */
+    policies: OnyxEntry<OnyxTypes.Policy>;
+    /** Beta features list */
+    betas: OnyxEntry<OnyxTypes.Beta[]>;
+    /** Indicated whether the report data is loading */
+    isLoadingReportData: OnyxEntry<boolean>;
+};
+
+type ComponentProps = OnyxProps & {
+    route: RouteProp<{params: {reportID: string}}>;
+};
+
+export default function <TProps extends ComponentProps, TRef>(WrappedComponent: ComponentType<TProps & RefAttributes<TRef>>) {
+    // eslint-disable-next-line rulesdir/no-negated-variables
+    function WithReportOrNotFound(props: TProps, ref: ForwardedRef<TRef>) {
+        const contentShown = React.useRef(false);
+
+        const shouldShowFullScreenLoadingIndicator = props.isLoadingReportData && (!Object.entries(props.report ?? {}).length || !props.report?.reportID);
+        // eslint-disable-next-line rulesdir/no-negated-variables
```

Comment:
Maybe let's disable this rule globally for this file? Top of the file:

```ts
/* eslint-disable rulesdir/no-negated-variables */
```

Reply:
Great idea, updated!